mqtt/mqtt-broker/src/configuration.rs

```diff
@@ -0,0 +1,298 @@
+            Duration::from_secs(90 * 24 * 60 * 60)
+        );
+    }
+
+    #[test]
+    fn it_refuses_persistence_with_no_file_path() {
+        let settings = BrokerConfig::from_file(Path::new("test/config_no_file_path.json"));
+
+        assert_matches!(settings, Err(_err));
+    }
+
+    #[test]
+    fn it_type_mismatch_fails() {
+        let settings = BrokerConfig::from_file(Path::new("test/config_bad_value_type.json"));
+
+        assert_matches!(settings, Err(_err));
+    }
+
+    #[derive(Debug, Deserialize)]
+    struct Container {
+        #[serde(deserialize_with = "humansize")]
+        size: u64,
+    }
+
+    #[test_case( "123b",  123 ; "when using bytes")]
+    #[test_case( "123kb",  123*1024 ; "when using kilobytes")]
+    #[test_case( "123mb",  123*1024*1024 ; "when using megabytes")]
+    #[test_case( "123gb",  123*1024*1024*1024 ; "when using gigabytes")]
+    fn it_deserializes_different_multipliers(input: &str, expected: u64) {
+        let container_json = json!({ "size": input }).to_string();
+
+        let container: Container = serde_json::from_str(&container_json).unwrap();
+        assert_eq!(container.size, expected);
+    }
+
+    #[test_case( "123kb",  123*1024 ; "when using all lowercase")]
+    #[test_case( "123Kb",  123*1024 ; "when using mixed case")]
+    #[test_case( "123KB",  123*1024 ; "when using all capitals")]
+    #[test_case( "  123kb",  123*1024 ; "when using leading spaces")]
+    #[test_case( "123kb  ",  123*1024 ; "when using trailing spaces")]
+    #[test_case( "123  kb",  123*1024 ; "when using separator spaces")]
+    #[test_case( "  123  kb  ",  123*1024 ; "when using spaces at multiple positions")]
+    fn it_deserializes_different_writing_modes(input: &str, expected: u64) {
+        let container_json = json!({ "size": input }).to_string();
+
+        let container: Container = serde_json::from_str(&container_json).unwrap();
+        assert_eq!(container.size, expected);
+    }
+
+    #[test_case( "123tb" ; "when using unknown unit")]
+    #[test_case( "123" ; "when missing unit")]
+    #[test_case( "12a3 kb" ; "when invalid number")]
+    fn it_fails_deserializing_unknown_unit(input: &str) {
+        let container_json = json!({ "size": input }).to_string();
+        let result = serde_json::from_str::<Container>(&container_json);
+
+        assert_matches!(result, Err(_err));
+    }
+
+    proptest! {
+        #[test]
+        fn doesnt_crash(s in "\\PC*") {
+            parse_size_anything(&s);
+        }
+    }
+
+    proptest! {
+        #[test]
+        fn it_parses_valid_input(lead in r"\s*",
+                                 num in any::<u64>(),
+                                 sep in r"\s*",
+                                 unit in r"(k|K|m|M|g|G)?(b|B)",
+                                 trail in r"\s*") {
+            let (input, expected) = get_input_and_expected(&lead, num, &sep, &unit, &trail);
+            parse_size_valid(&input, expected);
```

Comment:
I'd say lets move test body in the test function. Otherwise it's hard to read tests with usual structure Arrage, Act, Assert.
Here is a good example https://github.com/Azure/iotedge/blob/f3c7e9c8e7ace0db0390537ff5db1b8e324ffbf9/mqtt/mqtt-broker/src/subscription.rs#L297-L303

Reply:
You probably can 
1. create a custom type for all params. 
2. create a generator method
3. implement ToString for this type
4. almost copy-paste test from previous comment
....
the code will be much cleaner

Reply:
added a model structure for the test